language/compiler/ir_to_bytecode/syntax/src/ast.rs

```diff
@@ -143,13 +145,30 @@ pub enum StructDefinitionFields {
 #[derive(Debug, Eq, Hash, Ord, PartialEq, PartialOrd, Clone)]
 pub struct FunctionName(String);
 
+/// The kind of a type
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum Kind {
+    All,
+    Resource,
+    Unrestricted,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct KindConstraint(pub String, pub Kind);
+
+/// A list of kind constraints
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct KindConstraints(pub Vec<KindConstraint>);
```

Comment:
same as above

Reply:
However, I don't see the use of this particular newtype. Is there a situation in which a vector of `KindConstraint` should not be used here?

Reply:
Hmm... Forgot the reason why I decided to wrap it with a new type. I'll reconsider my design here while I do the renaming.

Reply:
When I wrote this code initially, I was wondering whether this should be a vec or a map, and since this type was referenced from many places in `compiler.rs`, I thought it would be better I give it an alias so that I only need to make one change in case I decide to do so.

Later it was clear to me since it needs to preserve the order it needs to be a vec.  Right now I'm inclining to change it to `Vec<KindConstraint>`. I'll have a second look into this.